src/settings/handlers/RoomAccountSettingsHandler.js

```diff
@@ -14,13 +15,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import SettingsHandler from "./SettingsHandler";
 import MatrixClientPeg from '../../MatrixClientPeg';
+import MatrixClientBackedSettingsHandler from "./MatrixClientBackedSettingsHandler";
+import {WatchManager} from "../WatchManager";
 
 /**
  * Gets and sets settings at the "room-account" level for the current user.
  */
-export default class RoomAccountSettingsHandler extends SettingsHandler {
+export default class RoomAccountSettingsHandler extends MatrixClientBackedSettingsHandler {
+    constructor() {
+        super();
+
+        this._watchers = new WatchManager();
+        this._onAccountData = this._onAccountData.bind(this);
+    }
+
+    initMatrixClient(oldClient, newClient) {
+        if (oldClient) {
+            oldClient.removeListener("Room.accountData", this._onAccountData);
+        }
+
+        newClient.on("Room.accountData", this._onAccountData);
+    }
+
+    _onAccountData(event, room) {
+        const roomId = room.roomId;
+
+        if (event.getType() === "org.matrix.room.preview_urls") {
+            let val = event.getContent()['disable'];
```

Comment:
Each exception to the general rule is slightly different enough to make that not worthwhile I think. Room previews are an inverted-if-boolean-but-null-otherwise, colors are an object, and everything else is a field on the content of a specific event. Mapping this out in a generic way would result in a lot of code which ultimately doesn't do much but prevent typos.